tools/swift-inspect/Sources/swift-inspect/WindowsRemoteProcess.swift

```diff
@@ -233,37 +237,266 @@ internal final class WindowsRemoteProcess: RemoteProcess {
+                                          readEventName)
+    if hReadEvent == HANDLE(bitPattern: 0) {
+      print("CreateEvent failed \(GetLastError())")
+      return
+    }
+    defer { CloseHandle(hReadEvent) }
+    let hWriteEvent: HANDLE = CreateEventA(LPSECURITY_ATTRIBUTES(bitPattern: 0),
+                                           false, // Auto-reset
+                                           false, // Initial state is nonsignaled
+                                           writeEventName)
+    if hWriteEvent == HANDLE(bitPattern: 0) {
+      print("CreateEvent failed \(GetLastError())")
+      return
+    }
+    defer { CloseHandle(hWriteEvent) }
+
+    // The path to the dll assuming it's in the same directory as swift-inspect.
+    let swiftInspectPath = ProcessInfo.processInfo.arguments[0]
+    let dllPath: String =
+        URL(fileURLWithPath: swiftInspectPath).deletingLastPathComponent().path +
+        "\\SwiftInspectClient.dll"
+    var dllPathRemote = UnsafeMutableRawPointer(bitPattern: 0)
+    var dllPathUtf16Len: UInt = 0
+    var dllPathWritten = false
+    dllPath.withCString(encodedAs: UTF16.self) {
+      // Check that the dll file exists
+      let dwAttrib = GetFileAttributesW($0)
+      if dwAttrib == INVALID_FILE_ATTRIBUTES ||
+         Int32(dwAttrib) & FILE_ATTRIBUTE_DIRECTORY != 0 {
+        print("\(dllPath) doesn't exist")
+        return
+      }
+      // Allocate memory and write dllPath in the remote process
+      dllPathUtf16Len = UInt(wcslen($0) * MemoryLayout<WCHAR>.size)
+      dllPathRemote = VirtualAllocEx(self.process,
+                                     nil,
+                                     SIZE_T(dllPathUtf16Len + 1), // +1 for `\0`
+                                     DWORD(MEM_COMMIT),
+                                     DWORD(PAGE_READWRITE))
+      if dllPathRemote == LPVOID(bitPattern: 0) {
+        print("VirtualAllocEx failed \(GetLastError())")
+        return
+      }
+      if !WriteProcessMemory(self.process, dllPathRemote, $0,
+                             SIZE_T(dllPathUtf16Len), nil) {
+        print("WriteProcessMemory failed \(GetLastError())")
+        return
+      }
+      dllPathWritten = true
+    }
+    if !dllPathWritten {
+      print("Failed to write dll path in the remote process")
       return
     }
 
-    repeat {
-      var entry: HEAPENTRY32 = HEAPENTRY32()
-      entry.dwSize = SIZE_T(MemoryLayout<HEAPENTRY32>.size)
+    var hKernel32 = HMODULE(bitPattern: 0)
+    "kernel32".withCString() {
+      hKernel32 = GetModuleHandleA($0)
+    }
 
-      if !Heap32First(&entry, dwProcessId, heap.th32HeapID) {
-        // FIXME(compnerd) log error
-        continue
+    // Load the dll and start the heap walk
+    let hookAddr = unsafeBitCast(GetProcAddress(hKernel32, "LoadLibraryW"),
+                                 to: LPTHREAD_START_ROUTINE.self)
+    let hThread: HANDLE = CreateRemoteThread(self.process, nil, 0, hookAddr,
```

Comment:
Mind blown that we can use `LoadLibraryW` as the entry point!

Reply:
Yeah, that was the magic trick that I had in mind for this.  The signature is proper, and `DllMain` with the flags allows us to inject execution nicely.